strawberry/types/fields/resolver.py

```diff
@@ -79,7 +79,9 @@ def type(self) -> Type[T]:
 
     @cached_property
     def is_async(self) -> bool:
-        return iscoroutinefunction(self.wrapped_func)
+        return iscoroutinefunction(self.wrapped_func) or isasyncgenfunction(
+            self.wrapped_func
+        )
```

Comment:
```suggestion
        return (
            iscoroutinefunction(self.wrapped_func)
            or isasyncgenfunction(self.wrapped_func)
        )
```
Nitpicky code cleanup

Reply:
what if I do
```suggestion
        return any(
             iscoroutinefunction(self.wrapped_func),
             isasyncgenfunction(self.wrapped_func)
        )
```

Just curios to know what you think 😊, btw will amend these when I fix the other subscription bug ;)

Reply:
I like that even better